chess.js

```diff
@@ -879,6 +879,22 @@ var Chess = function(fen) {
     return repetition
   }
 
+  function has_checkmate() {
+    var moves = generate_moves()
+    var winning_move = null
+    for (let index = 0; index < moves.length; index++) {
+      const move = moves[index]
+      make_move(move)
+      if (in_checkmate()) {
+        winning_move = move
+        undo_move()
```

Comment:
Remove this, you already have an undo_move call outside the if statement so you'll do an extra undo if you find a checkmate.

Reply:
Hi 
> Remove this, you already have an undo_move call outside the if statement so you'll do an extra undo if you find a checkmate.
thank you for code review.
You are right from performance point of view 
however I wanted  method not to modify chess object 
after calling method same side need to be on turn.
 
Also I return just one winning move : first found
potentially I could return array of  moves , if there are more 
moves to checkmate in one .  
potential improvement.
What do you think?

Reply:
Yes, I just saw that you have a break that breaks out of the for loop - code is fine! I don't know when/if this will get merged though - I am not sure this repo is maintained!

Reply:
me neither, but this is my 2 cents  ;-)